src/connectedvmware/azext_connectedvmware/custom.py

```diff
@@ -765,6 +768,174 @@ def get_hcrp_machine_id(
     return machine_id
 
 
+def reverseEndianLeftHalf(biosId: str):
+    """
+    Reverses the left half of the biosId.
+    This is required because the biosId format is different in HCRP Machine and InventoryItem.
+    Machine: 12345678-1234-1234-1234-123456789ABC
+    InventoryItem: 78563412-3412-3412-1234-123456789abc
+    """
+    def _rev(part: str):
+        return "".join(reversed([part[i:i + 2] for i in range(0, len(part), 2)]))
+    parts = biosId.split("-")
+    return "-".join(
+        [
+            _rev(parts[0]),
+            _rev(parts[1]),
+            _rev(parts[2]),
+            *parts[3:],
+        ]
+    )
+
+
+def create_from_machines(
+    cmd,
+    client: VirtualMachineInstancesOperations,
+    vcenter,
+    rg_name=None,
+    resource_name=None,
+):
+    vcenter_id = vcenter
+    machine_id = resource_name
+    if resource_name is not None:
+        if rg_name is None:
+            raise RequiredArgumentMissingError(
+                "--resource-group is required when --machine-name is provided."
+            )
+        machine_id = get_resource_id(
+            cmd,
+            rg_name,
+            HCRP_NAMESPACE,
+            MACHINES_RESOURCE_TYPE,
+            resource_name,
+        )
+    if not is_valid_resource_id(vcenter_id):
+        raise InvalidArgumentValueError(
+            "Please provide a valid vcenter resource id "
+            "using --vcenter-id."
+        )
+    assert isinstance(vcenter_id, str)
+
+    logger = get_logger(__name__)
+    arg_client = cf_resource_graph(cmd.cli_ctx)
+
+    vcenter_sub = vcenter_id.split("/")[2]
+    resources_client = get_resources_client(cmd.cli_ctx, vcenter_sub)
+    vcenter = resources_client.get_by_id(vcenter_id, VCENTER_KIND_GET_API_VERSION)
+    logger.info("Searching for machines in the vCenter %s ...", vcenter.name)
+
+    query = f"""
+Resources
+{rg_name and "| where resourceGroup == '{}'".format(rg_name) or ""}
+{machine_id and "| where id == '{}'".format(machine_id) or ""}
+| where type =~ 'Microsoft.HybridCompute/machines'
+| where isempty(kind) or kind =~ '{vcenter.kind}'
+| extend p=parse_json(properties)
+| extend u = tolower(tostring(p['vmUuid']))
+| where isnotempty(u)
+| where location =~ '{vcenter.location}'
+| extend vmUuidRev = strcat(
+    substring(u, 6, 2), substring(u, 4, 2), substring(u, 2, 2), substring(u, 0, 2), '-',
+    substring(u, 11, 2), substring(u, 9, 2), '-',
+    substring(u, 16, 2), substring(u, 14, 2), '-',
+    substring(u, 19))
+| project machineId=id, name, vmUuidRev
+| join kind=inner (
+ConnectedVMwareVsphereResources
+| where type =~ 'Microsoft.ConnectedVMwareVsphere/VCenters/InventoryItems'
+| where kind =~ 'VirtualMachine'
+| where id startswith '{vcenter.id}/InventoryItems'
```

Comment:
Is there any other way? Checking if it is  most efficeint way, we can also see portal ARG queries for reference...

Reply:
I didn't find any better approach.. went through the portal ARG queries too. [Link](https://msazure.visualstudio.com/One/_git/compute-hybrid-portal?path=/src/HybridComputeExtension/Client/React/Views/ArcPrivateCloud/VMware/Blades/VCenterVirtualMachines.ReactView.tsx&version=GC811e615ea4f4c65cc90392a3b2dcb8ae0f429e7f&line=529&lineEnd=530&lineStartColumn=17&lineEndColumn=120&lineStyle=plain&_a=contents)